blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1672,38 +1671,40 @@ fn check_block_messages<
     let mut check_msg = |msg: &UnsignedMessage,
                          account_sequences: &mut HashMap<Address, u64>,
                          tree: &StateTree<DB>|
-     -> Result<(), Box<dyn StdError>> {
+     -> Result<(), anyhow::Error> {
         // Phase 1: Syntactic validation
         let min_gas = price_list.on_chain_message(msg.marshal_cbor().unwrap().len());
-        msg.valid_for_block_inclusion(min_gas.total(), network_version)?;
+        msg.valid_for_block_inclusion(min_gas.total(), network_version)
+            .map_err(|e| anyhow::anyhow!("{}", e))?;
```

Comment:
Let's change `valid_for_block_inclusion` to use `anyhow::Error`.

Reply:
This doesn't work very well unfortunately.

Reply:
Oh? Can't you update `valid_for_block_inclusion` to use lots of `bail!`s?

Reply:
Indeed. Let's try again.